internal/cel/mutator.go

```diff
@@ -98,6 +110,30 @@ func mutate(pipelineRun *tekv1.PipelineRun, mutation *MutationRequest) *tekv1.Pi
 			pipelineRun.Annotations = make(map[string]string)
 		}
 		pipelineRun.Annotations[mutation.Key] = mutation.Value
+	case MutationTypeResource:
+		if pipelineRun.Annotations == nil {
+			pipelineRun.Annotations = make(map[string]string)
+		}
+
+		// Parse the new value as integer
+		newValue, err := strconv.Atoi(mutation.Value)
```

Comment:
in `createResourceMutationFunction` we are using `intValue, intValueOk := rhs.Value().(int64)`, I think we should prefer `strconv.ParseInt` to ensure int64 is used instead of `strconv.Atoi`.

```suggestion
		newValue, err := strconv.ParseInt(mutation.Value, 10, 64)
```